sympy/series/formal.py

```diff
@@ -0,0 +1,1119 @@
+    #  range of admissible functions.
+    if isinstance(f, Add):
+        result = False
+        ak = sequence(S.Zero, (0, oo))
+        ind, xk = S.Zero, None
+        for t in Add.make_args(f):
+            res = _compute_fps(t, x, 0, S.One, hyper, order, rational, full,
+                               False)
+            if res:
+                if not result:
+                    result = True
+                    xk = res[1]
+                if res[0].start > ak.start:
+                    seq = ak
+                    s, f = ak.start, res[0].start
+                else:
+                    seq = res[0]
+                    s, f = res[0].start, ak.start
+                save = Add(*[z[0]*z[1] for z in zip(seq[0:(f - s)], xk[s:f])])
+                ak += res[0]
+                ind += res[2] + save
+            else:
+                ind += t
+        if result:
+            return ak, xk, ind, mul
+        return None
+
+    result = None
+
+    # from here on it's x0=0 and dir=1 handling
+    k = Dummy('k')
+    if rational:
+        result = rational_algorithm(f, x, k, order, full)
+
+    if result is None and hyper:
+        result = hyper_algorithm(f, x, k, order)
+
+    if result is None:
+        return None
+
+    ak = sequence(result[0], (k, result[2], oo))
+    xk = sequence(x**k, (k, 0, oo))
+    ind = result[1]
+
+    return ak, xk, ind, mul
+
+
+def compute_fps(f, x, x0=0, dir=1, hyper=True, order=4, rational=True,
+                full=False):
+    """Computes the formula for Formal Power Series of a function.
+
+    Returns (sequence of coefficients, sequence of x, independent terms,
+    common terms).
+
+    Tries to compute the formula by applying the following techniques
+    (in order)
+
+    * rational_algorithm
+    * Hypergeomitric algorithm
+
+    Parameters
+    ==========
+
+    * x : Symbol, optional
+        If x is None and ``f`` is univariate, the univariate symbols will be
+        supplied, otherwise an error will be raised.
+
+    * x0 : numbers, optional
+        Point to perform series expansion about. Default is 0.
+
+    * dir : {1, -1, '+', '-'}, optional
+        If dir is 1 or '+' the series is calculated from the right and
+        for -1 or '-' the series is calculated from the left. For smooth
+        functions this flag will not alter the results. Default is 1.
+
+    * hyper : {True, False}, optional
+        Set hyper to False to skip the hypergeometric algorithm.
+        By default it is set to False.
+
+    * order : int, optional
+        Order of the derivative of ``f``, till which algorithms are run.
+        Default is 4.
+
+    * rational : {True, False}, optional
+        Set rational to False to skip rational algorithm. By default it is set
+        to True.
+
+    * full : {True, False}, optional
+        Set full to True to increase the range of rational algorithm.
+        See :func:`rational_algorithm` for details. By default it is set to
+        False.
```

Comment:
You're so close this time. Two things that are still wrong:

- Don't use `*` bullets.
- Don't leave a blank line between parameters.

A careful reading of the [documentation standard](https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt) really would help you find these mistakes.

Same thing goes for the other docstring below.